### PR TITLE
feat(tables): adding border prop

### PR DESCRIPTION
**Pull request by ccedrone**
Regarding: [774](https://github.com/cengage/react-magma/issues/774)

### CODE REVIEW

File: packages/react-magma-dom/src/components/Table/Table.stories.tsx [hunk 1]
```diff
@@ -89,10 +87,16 @@ export default {
 export const Default = Template.bind({});
 Default.args = {
   hasHoverStyles: false,
+  hasTableBorder: false,
```
Comment (silvalaura) [inline packages/react-magma-dom/src/components/Table/Table.stories.tsx:90]:
```suggestion
  hasOuterBorder: false,
```

File: packages/react-magma-dom/src/components/Table/Table.stories.tsx [hunk 2]
```diff
@@ -89,10 +87,16 @@ export default {
...
   hasVerticalBorders: false,
   hasZebraStripes: false,
 };
 
+export const WithTableBorder = Template.bind({});
+WithTableBorder.args = {
+  hasTableBorder: true,
+};
+
```
Comment (silvalaura) [inline packages/react-magma-dom/src/components/Table/Table.stories.tsx:99]:
```suggestion
export const WithOuterBorder = Template.bind({});
WithOuterBorder.args = {
  hasOuterBorder: true,
};

```

File: website/react-magma-docs/src/pages/api/table.mdx [hunk 3]
```diff
@@ -77,29 +76,104 @@ export function Example() {
...
+        ))}
+      </TableBody>
+    </Table>
+  );
+}
+```
+
+## Table With Border
+
+The `hasTableBorder` prop is a boolean which gives the table a border. By default the table border is off.
```
Comment (silvalaura) [inline website/react-magma-docs/src/pages/api/table.mdx:104]:
```suggestion
The `hasOuterBorder ` prop is a boolean which gives the table an outer border. By default the border is off.
```

File: packages/react-magma-dom/src/components/Table/Table.tsx [hunk 4]
```diff
@@ -16,6 +17,10 @@ export interface TableProps extends React.HTMLAttributes<HTMLTableElement> {
    * If true, row will be visually highlighted on hover
    */
   hasHoverStyles?: boolean;
+  /**
+   * If true, the table will have an outer border
```
Comment (silvalaura) [inline packages/react-magma-dom/src/components/Table/Table.tsx:21]: @default false

File: website/react-magma-docs/src/pages/api/table.mdx [hunk 5]
```diff
@@ -588,35 +640,89 @@ import {
...
   return (
-    <Card>
-      <Table>
+    <Card isInverse>
+      <Table
+        isInverse
+        hasZebraStripes
+        hasVerticalBorders
+        hasHoverStyles
+        hasTableBorder
```
Comment (silvalaura) [inline website/react-magma-docs/src/pages/api/table.mdx:655]:
```suggestion
        hasOuterBorder
```

File: website/react-magma-docs/src/pages/api/table.mdx [hunk 6]
```diff
@@ -77,29 +76,104 @@ export function Example() {
...
+    [
+      'Lorem ipsum dolor',
+      'Lorem ipsum dolor',
+      'Lorem ipsum dolor',
+      'Lorem ipsum',
+    ],
+  ];
+
+  return (
+    <Table hasTableBorder>
```
Comment (silvalaura) [inline website/react-magma-docs/src/pages/api/table.mdx:158]:
```suggestion
    <Table hasOuterBorder>
```

**Review by silvalaura — APPROVED** [review]

**Comment by silvalaura** [comment]
As mentioned in the team slack channel, we also want the table to have rounded corners by default. We will also add a `hasSquareCorners` prop.

File: packages/react-magma-dom/src/components/Table/Table.tsx [hunk 7]
```diff
@@ -16,9 +17,19 @@ export interface TableProps extends React.HTMLAttributes<HTMLTableElement> {
...
+  hasOuterBorder?: boolean;
   /**
    * If true, columns will have vertical borders
    */
+  hasSquareCorners?: boolean;
+  /**
+   * If true, the table will have square edges
+   * @default false
+   */
   hasVerticalBorders?: boolean;
```
Comment (silvalaura) [inline packages/react-magma-dom/src/components/Table/Table.tsx:33]: The descriptions are swapped here for the props

**Review by silvalaura — APPROVED** [review]

**Review by orion-cengage — APPROVED** [review]

**Pull request merged**